src/Components/CreateImageWizard/CreateImageWizard.js

```diff
@@ -517,12 +514,32 @@ const formStepHistory = (composeRequest) => {
 const CreateImageWizard = () => {
   const dispatch = useDispatch();
   const navigate = useNavigate();
-  const location = useLocation();
+  // composeId is an optional param that is used for Recreate image
+  const { composeId } = useParams();
+
+  // This is a bit awkward, but will be replaced with an RTKQ hook very soon
+  // We use useStore() instead of useSelector() because we do not want changes to
+  // the store to cause re-renders, as the composeId (if present) will never change
+  const { getState } = useStore();
+  const compose = getState().composes?.byId?.[composeId];
+  const composeRequest = compose?.request;
+
+  // TODO: This causes an annoying re-render when using Recreate image
```

Comment:
Not a TODO anymore, right? :tada:

Reply:
A different re-render. My bad.

Reply:
I wish! :smile: Still going to be an issue for the moment, we will get to it soon.